bootstrap/kubeadm/Makefile

```diff
@@ -1,239 +0,0 @@
-# Copyright 2019 The Kubernetes Authors.
```

Comment:
Are we losing any targets or validations that would be helpful to the larger project?

Reply:
The Makefile is still available in the old repository, we can port as needed. I didn't miss any targets for basic interactions

Reply:
I'd prefer not defaulting to untracked work to be done at a later time, can you create an issue with the actual differences that exist to resolve?

Reply:
There is really nothing that's particular to CABPK in the repository today

Reply:
Just a quick comparison uncovers these differences to me:
- `make test` generates a coverage profile in CABPK, but not CAPI
- the `lint` and `lint-full` targets different golang-ci configs (but that can be tracked via an issue specific for golang-ci config)
- CABPK has a `run` target for just running the manager binary
- CABPK also has `install` and `deploy` targets that are not present in CAPI

Reply:
If we are removing previous functionality from dev workflow, then we should have buy in for this from the `cluster-api-bootstrap-provider-kubeadm-maintainers` group, since they have delegated ownership of this part of the project.